torch/nn/utils/spectral_norm.py

```diff
@@ -40,8 +43,7 @@ def remove(self, module):
     def __call__(self, module, inputs):
         weight, u = self.compute_weight(module)
         setattr(module, self.name, weight)
-        with torch.no_grad():
-            getattr(module, self.name).copy_(weight)
+        setattr(module, self.name + '_u', u)
```

Comment:
Uh, yes, this needs to be `_u`, not `weight`. I seem to recall that there was a reason to use `copy_`, though. My impression is that this might behave differently across Python versions, in particular Python 3.6 vs. earlier. Is it OK to overwrite a registered buffer with a calculated variable?

Reply:
It should be okay. The `__setattr__` removes the current assigned values first anyways. According to local test with Py 3.6, it seems not leaking.

Reply:
Right. Thanks for checking. (What I was concerned about was whether there is an analogous issue to [the "cannot overwrite parameter with variable check"](https://github.com/pytorch/pytorch/blob/master/torch/nn/modules/module.py#L549) only working in Python 3.6. But it's not the case here.